server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java

```diff
@@ -345,17 +358,426 @@ public void deleteRegisteredModel(String fullName, boolean force) {
 
   public void deleteRegisteredModel(
       Session session, UUID schemaId, String registeredModelName, boolean force) {
-    if (force) {
-      // Remove all model versions
-    } else {
-      // Check if model versions exist and throw with unable to delete if they exist
-    }
     RegisteredModelInfoDAO registeredModelInfoDAO =
         getRegisteredModelDao(session, schemaId, registeredModelName);
     if (registeredModelInfoDAO == null) {
       throw new BaseException(
           ErrorCode.NOT_FOUND, "Registered model not found: " + registeredModelName);
     }
+    if (force) {
+      // Remove all model versions
+      List<ModelVersionInfoDAO> versionList =
+          getModelVersionsDao(session, registeredModelInfoDAO.getId(), "0", 100);
```

Comment:
Not sure how we should allow the `100` here to be parameterized.  This is purely a `keep pulling back versions to delete until we have deleted everything` operation and the number shouldn't really be user controlled, but you also don't want to pull back potentially 1000s of records.

Reply:
Let's keep the behaviour same as other places where we are deleting child entities. Also can we move `100` to a local variable.

Reply:
It's hard coded to 1 in other places which doesn't seem like the best way to tackle a bulk delete.  I'll update to 1, but we should really look at increasing the max limit (or at least using the default max limit) of the list operations.